File: Go/Structural/Bridge/ReportEngines/jsonEngine.go
package ReportEngines

import (
	"github.com/MastersAcademy/ma_ap_2016/Go/Structural/Bridge/Models"
	"encoding/json"
	"log"
)

type JsonEngine struct {

}

func NewJsonEngine() ReportEngine {
	return &JsonEngine{}
}

func (e *JsonEngine) BuildReport(videos []Models.Video) string  {
	b, err := json.Marshal(videos)
	if err != nil {
		log.Panic("can't serialize json")
	}
	return string(b)
}

File: Swift/Structural/Bridge/Bridge/JSONEngine.swift
//
//  JSONEngine.swift
//  Bridge
//
//  Created by Anton on 26.01.17.
//  Copyright © 2017 None. All rights reserved.
//

import Foundation

class JSONEngine: ReportEngineProtocol {
    func buildReport(videos: [Video]) -> String {
        return Json.jsonString(from: videos)
    }
}
